udata/migrations/2016-01-12-add-tag-slug.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Add the slug property to Tags with a default value based on the name
+ * property.
+ */
+
+ // Taken from underscore.string
+function defaultToWhiteSpace(characters) {
+  if (characters == null)
+    return '\\s';
+  else if (characters.source)
+    return characters.source;
+  else {
+    characters = characters
+      ? String(characters).replace(/([.*+?^=!:${}()|[\]\/\\])/g, '\\$1')
+      : '';
+
+    return '[' + characters + ']';
+  }
+}
+
+// Taken from underscore.string
+function slugify(str) {
+  var from = 'ąàáäâãåæăćęèéëêìíïîłńòóöôõøśșțùúüûñçżź';
+  var to = 'aaaaaaaaaceeeeeiiiilnoooooosstuuuunczz';
+  var regex = new RegExp(defaultToWhiteSpace(from), 'g');
+
+  str = String(str).toLowerCase().replace(regex, function(c) {
+    var index = from.indexOf(c);
+    return to.charAt(index) || '-';
+  });
+
+  str = str.replace(/[^\w\s-]/g, '');
+  str = str.trim();
+  str = str.replace(/([A-Z])/g, '-$1');
+  str = str.replace(/[-_\s]+/g, '-');
+  str = str.toLowerCase();
+  return str;
+}
+
+var tags = db.tag.find();
+var count = 0;
+
+tags.forEach(function(tag) {
+  tag.slug = slugify(tag.name);
+  db.tag.save(tag);
```

Comment:
I wonder if re-saving each tag in Python once the new field is added with both `populate_from` and `required` is less prone to bugs.